rtengine/iplocallab.cc

```diff
@@ -14276,13 +14278,50 @@ void ImProcFunctions::Lab_Local(
         return;
     }
 
-    //BENCHFUN
+
 
     constexpr int del = 3; // to avoid crash with [loy - begy] and [lox - begx] and bfh bfw  // with gtk2 [loy - begy-1] [lox - begx -1 ] and del = 1
     struct local_params lp;
     calcLocalParams(sp, oW, oH, params->locallab, lp, prevDeltaE, llColorMask, llColorMaskinv, llExpMask, llExpMaskinv, llSHMask, llSHMaskinv, llvibMask, lllcMask, llsharMask, llcbMask, llretiMask, llsoftMask, lltmMask, llblMask, lllogMask, ll_Mask, llcieMask, locwavCurveden, locwavdenutili);
 
     //avoidcolshi(lp, sp, transformed, reserved,  cy, cx, sk);
+    //BENCHFUN
+
+
+
+
+    //avoid negative values RGB then Lab when using before SE CBDL or Dehaze, or processor type...
+
+    int bw0 = transformed->W;
+    int bh0 = transformed->H;
+
+    float epsi0 = 0.000001f;
```

Comment:
This means it avoids negative and zero values. Should the setting be renamed to "Avoid zero and negative values" (or "Avoid non-positive values")?

Reply:
I will push "Avoid zero and negative values"
Just a note to show the specificity of languages ​​in French, it is a mistake not to put an "s" in "negative"

Reply:
Is it because adjectives in French must have an extra -s when they refer to plural nouns? In English, adjectives don't have plural forms. Spell check might not catch "negatives" because "negative" can also be a noun.

Reply:
I am not disputing the grammar of another language, just whether these "subtleties" are well perceived by automatic translators...

Reply:
I know. I'm just asking out of curiosity. Besides English, I have only ever learned (some) Spanish. Spanish has plural forms of adjective and I was wondering if it's the same in French.

Reply:
French is a complex language. There are several types of adjectives. The agreement of the conjunctions depends on the auxiliary verb and the position of the object complement.
It is a very old language taking its sources from Latin, Greek, a (little) bit of Germanic languages, Arabic... and now recently English (through social networks, students). But did you know that about 50% of English words come from French (and not the other way around).